### PR TITLE
[6.7] Zoom-out (contentOnly): show more menu in block toolbar

### LINKED ISSUE #66287: Restore ability to copy and delete patterns in zoom out
Now that we reverted the [vertical toolbar and have returned to the regular toolbar](https://github.com/WordPress/gutenberg/pull/66039), some of us were testing zoom out today and noticed that we missed having a quick "trash" option. Without it, it's not exactly clear how to remove a pattern, even though using the delete key on the keyboard does still remove the pattern. Considering the prior vertical toolbar included a trash button, I'm marking this as a bug.

### PR DESCRIPTION
**Pull request by ellatrix**
## What?

Since zoom-out mode is automatically `contentOnly` mode, this condition will cause the "more" menu in the block toolbar to disappear. It's actually shown in trunk and was fixed here:

https://github.com/WordPress/gutenberg/pull/65485/files#diff-9ac2a4466e7c0a85aaff0b40c5d7bc64c894132528ca3c879db4bebb438617dcL245-L247

I do wonder if we should try to backport all of #65485 or not. Cc @youknowriad.

Also, since this is `contentOnly` mode, should the group/ungroup buttons be removed? I went ahead and removed them, but not sure if they were left in place intentionally.

## Why?

There's no way to trash patterns in zoom-out mode in 6.7 without the use of the keyboard. Fixes https://github.com/WordPress/gutenberg/issues/66287

## How?

Remove the requirement to be in the default editing mode.

## Testing Instructions

- Engage Zoom Out
- Click on a section
- Toggle the "3 dots" block settings menu
- Check the options are stable for Zoom Out (should be primarily delete and copy)
- Repeat above for:
    - Pages in the _Site_ Editor
    - Pages in the _Post_ Editor

### CODE REVIEW

**Comment by annezazu** [comment]
Could we just show duplicate and delete? I find having group, ungroup, and create pattern to be an odd experience, in particular group/ungroup as nothing changes visually:

https://github.com/user-attachments/assets/b334507b-a3a9-493f-9455-d5ef46e9bfba

**Comment by ellatrix** [comment]
@annezazu I had just deleted them yes.

**Comment by annezazu** [comment]
Hmm noticing an empty menu randomly. I'm not quite sure why it appears but I can replicate it if I play around with this PR enough:

Screenshot 2024-10-22 at 11 54 04 AM

Here's a quick video where this showed up:

https://github.com/user-attachments/assets/3764af33-f5c7-4c1d-a706-d46101fcfa82

**Comment by richtabor** [comment]
I'd expect "Copy" as well. 

Perhaps "add before" and "add after", though those could be separate—to engage with the zoom out behavior that occurs when using one of those inserters. 

This is what I'd consider ideal: 

CleanShot 2024-10-22 at 11 56 56

**Comment by ellatrix** [comment]
> Hmm noticing an empty menu randomly. I'm not quite sure why it appears but I can replicate it if I play around with this PR enough:

I suspect it's from the grouping you did, which is now gone. Does it still appear after that change?

**Comment by ellatrix** [comment]
@richtabor Copy was intentionally removed in #61127. Are you sure we should restore it here?

**Review by getdave — COMMENTED** [review]
Testing on a `Page` I saw options relating to `Edit template`. However the block I'm interacting with is _not_ part of the template.


Screen Shot 2024-10-22 at 10 59 11

This also happens when selecting full width sections like the yellow "Stories" section below.

**Comment by ellatrix** [comment]
@getdave Does this appear in trunk as well? What content do you have to reproduce it?

**Comment by ellatrix** [comment]
Ah, I see it on the demo page actually, sorry.

**Comment by draganescu** [comment]
We ehave no guarantee what kind of selection this applies to so the contents of the settings dropdown varies a lot.

**Comment by ellatrix** [comment]
@getdave So the reason you see that button is because #65560 (from @youknowriad) was not backported.

**Comment by annezazu** [comment]
I'm still seeing "create pattern" as an option.

>I suspect it's from the grouping you did, which is now gone. Does it still appear after that change?

Can't replicate anymore 🥳

**Comment by getdave** [comment]
> @getdave So the reason you see that button is because #65560 (from @youknowriad) was not backported.

Looks like we now have that PR backported in https://github.com/WordPress/gutenberg/pull/66339.

**Comment by getdave** [comment]
Testing _locally_ with [the backport](https://github.com/WordPress/gutenberg/pull/66339) applied I'm still seeing the rogue menu items when editing `Pages` in the Site Editor.

These items are adding a lot of additional complexity to Zoom Out mode in 6.7. As such I"m bumping up the priority on this one with a view to finding a fix before RC2 next week.

I think we should be aiming for this UI which also means getting rid of the `Create Pattern` menu item:

Screenshot 2024-10-24 at 10 02 16

**Comment by talldan** [comment]
> Testing on a Page I saw options relating to Edit template. However the block I'm interacting with is not part of the template. 

I've pushed a commit that may solve that (feel free to do what you want with it!), essentially not showing those options for any children of the post content block. I think this makes sense, as the children of Post Content are not ever considered part of the template AFAIK.

I'm still seeing 'Create pattern' a lot when testing.

**Comment by getdave** [comment]
Thanks @talldan. This does indeed solve the Issue of the unwanted "Template" items in the Site Editor.

Unfortunately it does not seem to fix the Post Editor (try editing the same `Page` in the Post Editor). I still see the Template items which is of course expected because the `core/post-content` block will not exist in the Post Editor.

Screen Shot 2024-10-24 at 12 05 38


I'm wonder whether we just skip rendering these items in Zoom Out mode? It's unlikely we'd ever want them. I feel like this is a valid case to tie this logic to.


------

Also, @richtabor how do you feel about getting ride of the `Create Pattern` option only in Zoom Out? I don't want this question to block merging this PR but if we can tackle two problems with one PR that might also be nice.

**Comment by annezazu** [comment]
Noted above on my end but I’d like create pattern to go away personally. It’s a confusing UX as is if you click it.

**Comment by ellatrix** [comment]
Why didn't we rebase this first?

**Comment by talldan** [comment]
Given @getdave's commit, I'll revert my one.

**Comment by talldan** [comment]
> Why didn't we rebase this first?

Now done. Rather than revert I dropped the commit I added, as it meant not solving a conflict.

Was there some other change in wp/6.7 that would be beneficial?

**Comment by getdave** [comment]
> Was there some other change in wp/6.7 that would be beneficial?

I think https://github.com/WordPress/gutenberg/pull/66339

**Comment by ellatrix** [comment]
Well, looking at the latest changes that component is now entirely hidden? Is that needed after https://github.com/WordPress/gutenberg/pull/66339?

**Comment by getdave** [comment]
> Well, looking at the latest changes that component is now entirely hidden? Is that needed after #66339?

I rebase against latest `wp/6.7`. If you reset the HEAD to remove my last commit you'll see it still doesn't fix the problem when editing `Pages` 😬

File: packages/editor/src/components/provider/index.js
Comment (getdave) [inline packages/editor/src/components/provider/index.js]: Would it be better to subscribe to state only in the `ContentOnlySettingsMenu`? That would isolate the subscription and also keep it closer to the affected UI.

The downside would be that any subscriptions in `ContentOnlySettingsMenu` would be run.

**Comment by richtabor** [comment]
> Also, @richtabor how do you feel about getting ride of the Create Pattern option only in Zoom Out? I don't want this question to block merging this PR but if we can tackle two problems with one PR that might also be nice.

Yes, let's take it out.

**Comment by draganescu** [comment]
I've added copy back and removed the pattern related group of options.

**Comment by kevin940726** [comment]
Should this be targeted at 6.7? Technically, it's a feature, and we're already in the RC phase. Could we punt this to 6.7.1 or 6.8? Sorry if I'm missing something! c.c. @colorful-tones @ndiego

**Comment by annezazu** [comment]
The main piece we need for 6.7 is the ability to delete using the more menu. I say this needs to land for the release.

**Review by andrewserong — APPROVED** [review]
> The main piece we need for 6.7 is the ability to delete using the more menu. I say this needs to land for the release.

I agree. It can be tricky this close to the last RC before the final release, but from my perspective this would be good to get in:

* The change is relatively small and doesn't feel too dangerous to land to me
* The ability to delete from the block settings menu is important enough to warrant this change
* Being able to copy, duplicate, and delete feels like a good simple set of controls to me
* It's testing well for me in the site editor, page editor, and post editor in zoom out mode, and I haven't encountered any issues not in zoom out

image

I only had a small amount of time to test this before wrapping up for the day, so it might be worth getting a second review on this, but it LGTM and I think would be good to get in for 6.7 if we can.

**Comment by kevin940726** [comment]
I rebased the branch to see if it passes the CI.

**Review by kevin940726 — APPROVED** [review]
It's also testing well for me. Let's get this in for RC 3.

**Comment by richtabor** [comment]
We also need this merged addressed in `trunk`.

**Pull request merged**
